go/src/url_shortener/url_shortener.go

```diff
@@ -114,8 +123,52 @@ func (c *URLShortener) shortenHandler(w http.ResponseWriter, r *http.Request) {
 	c.incrementHandlerCounter(c.shortenRoute, true)
 }
 
+type redirectsJSON struct {
+	Success int `json:"success"`
+	Failed  int `json:"failed"`
+}
+
+type handlerJSON struct {
+	Name  string `json:"name"`
+	Count int    `json:"count"`
+}
+
+type serverStatsJSON struct {
+	TotalURL  int           `json:"total_url"`
+	Redirects redirectsJSON `json:"redirects"`
+	Handlers  []handlerJSON `json:"handlers"`
+}
+
+type statsJSON struct {
+	ServerStats serverStatsJSON `json:"server_stats"`
+}
```

Comment:
At the moment, each time the statistics handler is called we need to recreate the structures in order to serialize to JSON.

Can we try to use the structures highlighted in this comment to keep track of the statistics? and them remove the current fields of  `urlStatistics`.